Simplified/NYPLSettingsAccountDetailViewController.m

```diff
@@ -45,66 +45,81 @@ typedef NS_ENUM(NSInteger, CellKind) {
 
 @interface NYPLSettingsAccountDetailViewController () <NYPLUserAccountInputProvider, NYPLSettingsAccountUIDelegate>
 
+// State machine
 @property (nonatomic) BOOL isLoggingInAfterSignUp;
 @property (nonatomic) BOOL loggingInAfterBarcodeScan;
+@property (nonatomic) BOOL loading;
+@property (nonatomic) BOOL hiddenPIN;
+
+// UI
 @property (nonatomic) UIImageView *barcodeImageView;
 @property (nonatomic) UILabel *barcodeImageLabel;
 @property (nonatomic) NSLayoutConstraint *barcodeHeightConstraint;
 @property (nonatomic) NSLayoutConstraint *barcodeLabelSpaceConstraint;
 @property (nonatomic) float userBrightnessSetting;
-
 @property (nonatomic) NSMutableArray *tableData;
-@property (nonatomic, copy) void (^completionHandler)(void);
-@property (nonatomic) BOOL hiddenPIN;
 @property (nonatomic) UIButton *PINShowHideButton;
 @property (nonatomic) UIButton *barcodeScanButton;
-@property (nonatomic) NSString *selectedAccountId;
-@property (nonatomic) Account *selectedAccount;
-@property NYPLUserAccountFrontEndValidation *frontEndValidator;
-
-@property (nonatomic) BOOL loading;
-
 @property (nonatomic) UITableViewCell *registrationCell;
 @property (nonatomic) UITableViewCell *logInSignOutCell;
 @property (nonatomic) UITableViewCell *ageCheckCell;
-
 @property (nonatomic) UISwitch *syncSwitch;
-@property (nonatomic) BOOL permissionCheckIsInProgress;
 
+// account state
+@property NYPLUserAccountFrontEndValidation *frontEndValidator;
+@property (nonatomic) NYPLSignInBusinessLogic *businessLogic;
+
+// networking
 @property (nonatomic) NSURLSession *session;
 @property (nonatomic) NYPLSettingsAccountURLSessionChallengeHandler *urlSessionDelegate;
 
 @end
 
+static const NSInteger sLinearViewTag = 1;
+static const CGFloat sVerticalMarginPadding = 2.0;
+static const double sRequestTimeoutInterval = 25.0;
+
+// table view sections indeces
+static const NSInteger sSection0AccountInfo = 0;
+static const NSInteger sSection1Sync = 1;
 
 @implementation NYPLSettingsAccountDetailViewController
 
 @synthesize usernameTextField;
 @synthesize PINTextField;
 
-NSInteger const linearViewTag = 1;
-CGFloat const verticalMarginPadding = 2.0;
-double const requestTimeoutInterval = 25.0;
+#pragma mark - Computed variables
+
+- (NSString *)selectedAccountId
+{
+  return self.businessLogic.libraryAccountID;
+}
+
+- (Account *)selectedAccount
+{
+  return self.businessLogic.libraryAccount;
+}
 
 - (NYPLUserAccount *)selectedUserAccount
 {
-  return [NYPLUserAccount sharedAccount:self.selectedAccountId];
+  return self.businessLogic.userAccount;
 }
 
-#pragma mark NSObject
+#pragma mark - NSObject
 
-- (instancetype)initWithAccount:(NSString *)account
+// overriding superclass's designated initializer
+- (instancetype)initWithStyle:(__unused UITableViewStyle)style
 {
-  self.selectedAccountId = account;
-  self.selectedAccount = [[AccountsManager sharedInstance] account:self.selectedAccountId];
-  return [self init];
+  return [self initWithAccount:nil];
```

Comment:
Is this intended? Cause the `nil` value will be passed into `[[NYPLSignInBusinessLogic alloc] initWithLibraryAccountID:libraryUUID]` and it's not optional on the Swift side

Reply:
that's a good point, I'll take a look at what I can do. Maybe I'll have to disable that initializer, although I'm supposed to override it per Cocoa conventions.

Reply:
I pushed a change where I ended up continuing the initialization as it was done in other parts of the code (see NYPLSettingsAccountsList.swift). This should be fine because there's no code path actually invoking `initWithStyle:` for this VC. However, I also added an assert to catch this situation if we incur in it in the future.
Thanks for catching this.